pkg/request/batchstore.go

```diff
@@ -7,36 +7,34 @@ package request
 
 import (
 	"context"
-	"fmt"
 	"sync"
 	"sync/atomic"
-	"time"
 )
 
 type BatchStore struct {
-	batchingEnabled uint32
-	currentBatch    *batch
-	readyBatches    []*batch
-	onDelete        func(key string)
-	batchMaxSize    uint32
-	maxCapacity     int
-	keys2Batches    sync.Map
-	lock            sync.RWMutex
-	signal          sync.Cond
+	currentBatch      *batch
+	readyBatches      []*batch
+	onDelete          func(key string)
+	batchMaxSize      uint64
```

Comment:
1. why did we change it to uint64? Do you think 32 bits is not enough? 

2. Let's make both these uint64 limits - uint32, and then we can store them in a single uint64. This way, we can read them both atomically in a single operation.